### DISCUSSION TITLE
imgui render to texture

### DISCUSSION THREAD

**Discussion by matan45** [ocornut/imgui#6322] [New Users Build/Link/Run issues ONLY!]
hi i am trying to render to texture with imgui from this iusse https://github.com/ocornut/imgui/issues/5110
and i getting this result 
![image](https://user-images.githubusercontent.com/27070676/231375537-ac58e374-ad42-4b71-ba01-e673d72095d6.png)
the question is why i dont see the clear color i set the on the rendering pass 

	
            ` vk::ClearValue clearColor = { std::array<float, 4>{0.0f, 1.0f, 1.0f, 0.0f} };

		vk::RenderingAttachmentInfoKHR colorAttachment{};
		colorAttachment.imageView = imageFrame.viewPortImageView;
		colorAttachment.imageLayout = vk::ImageLayout::eAttachmentOptimalKHR;
		colorAttachment.loadOp = vk::AttachmentLoadOp::eClear;
		colorAttachment.storeOp = vk::AttachmentStoreOp::eStore;
		colorAttachment.clearValue = clearColor;

		vk::RenderingAttachmentInfoKHR depthStencilAttachment{};
		depthStencilAttachment.imageView = depthStencil.viewPortImageView;
		depthStencilAttachment.imageLayout = vk::ImageLayout::eDepthAttachmentOptimalKHR;
		depthStencilAttachment.loadOp = vk::AttachmentLoadOp::eClear;
		depthStencilAttachment.storeOp = vk::AttachmentStoreOp::eStore;
		depthStencilAttachment.clearValue.depthStencil.depth = 1.0f;
		depthStencilAttachment.clearValue.depthStencil.stencil = 0;

		vk::RenderingInfoKHR renderPassInfo{};
		renderPassInfo.sType = vk::StructureType::eRenderingInfoKHR;
		renderPassInfo.renderArea = { 0, 0, swapchainExtent.width, swapchainExtent.height };
		renderPassInfo.layerCount = 1;
		renderPassInfo.colorAttachmentCount = 1;
		renderPassInfo.pColorAttachments = &colorAttachment;
		renderPassInfo.pDepthAttachment = &depthStencilAttachment;
		renderPassInfo.pStencilAttachment = &depthStencilAttachment;

		commandBuffer.beginRenderingKHR<vk::DispatchLoaderDynamic>(&renderPassInfo, dldi);

		commandBuffer.bindPipeline(vk::PipelineBindPoint::eGraphics, engine->getShader(shaderIndex)->getPipeline());

		vk::Viewport viewport{};
		viewport.x = 0.0f;
		viewport.y = 0.0f;
		viewport.width = static_cast<float>(swapchainExtent.width);
		viewport.height = static_cast<float>(swapchainExtent.height);
		viewport.minDepth = 0.0f;
		viewport.maxDepth = 1.0f;
		commandBuffer.setViewport(0, 1, &viewport);

		vk::Rect2D scissor{};
		scissor.offset.x = 0;
		scissor.offset.y = 0;
		scissor.extent = swapchainExtent;
		commandBuffer.setScissor(0, 1, &scissor);

		std::array<vk::Buffer, 1> vertexBuffers = { vertexBuffer->getBuffer() };
		std::array<vk::DeviceSize, 1> offsets = { 0 };
		commandBuffer.bindVertexBuffers(0, static_cast<uint32_t>(vertexBuffers.size()), vertexBuffers.data(), offsets.data());

		commandBuffer.bindIndexBuffer(indiceBuffer->getBuffer(), 0, vk::IndexType::eUint16);

		std::array<vk::DescriptorSet, 1> descriptorSets = { uniformBuffers->getDescriptorSets()[0] };
		commandBuffer.bindDescriptorSets(vk::PipelineBindPoint::eGraphics, engine->getShader(shaderIndex)->getPipelineLayout(), 0, descriptorSets, nullptr);

		commandBuffer.drawIndexed(static_cast<uint32_t>(indices.size()), 1, 0, 0, 0);

		commandBuffer.endRenderingKHR<vk::DispatchLoaderDynamic>(dldi);`

note i also use the vulkan extension VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME

**Comment by ocornut** [comment]
Hello,
Do you realize this is 100% a Vulkan question and not a Dear ImGui question?
Consider using RenderDoc (gpu debugger) to understand what is going on.
